code/headers/module.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <base_module.hpp>
+#include <lidar.hpp>
+
+// This is the main module class that measures the smallest distance and sends
+// it on the canbus
+namespace r2d2::distance {
+
+    /// \brief The main class for the distance module.
+    /// \detail
+    /// This class uses a lidar to measure the distance between the lidar and
+    /// the closest object. When a distance request has been received this class
+    /// sends the smallest detected distance over the can-bus.
+    class module_c : public base_module_c {
+    private:
+        lidar_c& lidar;
+
+        uint16_t smallest_value;
+        uint16_t current_value;
+        bool smallest_value_set_first_time = false;
```

Comment:
Tried to understand this bool, not sure what it does, sure it works, care to explain?

Reply:
The module sends the smallest distance value on the canbus. To do this, all measured distances are compared with the last smallest distance encountered. This however raises the problem of what the initial smallest value is, before any measurements are made. If the largest possible value is used, what happens when only failed measurements (with a value of 0) are received? Then this largest value will be send on the canbus, while that is not a valid measurement. To make sure only a valid measurement is used in the comparison, a bool, ```smallest_value_set_first_time ```, is used to keep track of whether a valid measurement has been received that can be used in the comparison.

Reply:
Makes sense, thank you for the explanation!